### PR TITLE
py-phydms, py-pypdf2, and py-pyvolve: new packages

### PR DESCRIPTION
**Pull request by luke-dt**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-phydms/package.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+       codon models (ExpCM) for phylogenetic inference and the detection of
+       biologically interesting selection."""
+
+    homepage = "http://jbloomlab.github.io/phydms"
+    pypi     = "phydms/phydms-2.4.1.tar.gz"
+
+    version('2.4.1', sha256='04eb50bdb07907214050d19214d9bc8cf2002e24ca30fbe6e0f23f013d584d5c')
+
+    depends_on('python@3.5:',   type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-phydms/package.py:22]: Missing dependencies:
```python
    setup_requires = [                                                                   
        'cython>=0.28',                                                                  
        'numpy>=1.11',                                                                   
        ],                                                                               
    install_requires = [                                                                 
        'biopython>=1.67',                                                               
        'cython>=0.28',                                                                  
        'numpy>=1.16.5',                                                                 
        'scipy>=0.18',                                                                   
        'matplotlib>=2.0.2',                                                             
        'natsort>=5.0.1',                                                                
        'sympy>=1.0',                                                                    
        'six>=1.10',                                                                     
        'pandas>=0.20.2',                                                                
        'pyvolve>=1.0.3',                                                                
        'statsmodels>=0.8',                                                              
        'weblogo>=3.4, <3.6',                                                            
        'PyPDF2>=1.26',                                                                  
        ],                  
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-phydms/package.py (hunk 1 above)
Comment (luke-dt) [reply var/spack/repos/builtin/packages/py-phydms/package.py:22]: woops! this is a dependency of another package I'm working on which has the same dependencies, so I accidentally overlooked this.

File: var/spack/repos/builtin/packages/py-pypdf2/package.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+       It can also add custom data, viewing options, and passwords to PDF files.
+       PyPDF2 can retrieve text and metadata from PDFs as well."""
+
+    homepage = "https://pypdf2.readthedocs.io/en/latest/"
+    pypi     = "PyPDF2/PyPDF2-2.5.0.tar.gz"
+
+    version('2.5.0',  sha256='5802b1f40fa79be1b5ab9edc95a4e7f7e73399589db4f0e66ca831f449e7a2cd')
+    version('1.26.0', sha256='e28f902f2f0a1603ea95ebe21dff311ef09be3d0f0ef29a3e44a932729564385')
+
+    depends_on('python@3.6:3.9',       type=('build', 'run'), when='@2.0.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pypdf2/package.py]:
```suggestion
    depends_on('python@3.6:',       type=('build', 'run'), when='@2.0.0:')
```

File: var/spack/repos/builtin/packages/py-pypdf2/package.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on('py-setuptools',        type='build')
+    depends_on('py-typing-extensions', type=('build', 'run'), when='@2.0.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pypdf2/package.py]:
```suggestion
    depends_on('py-typing-extensions', type=('build', 'run'), when='@2.0.0: ^python@:3.9')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**